test/functional/sqlsrv/srv_570_fetch_varbinary.phpt

```diff
@@ -0,0 +1,146 @@
+{
+    while (($line1 = fread($input, 80)) && ($line2 = fread($result, 80))) {
+        if ($line1 != $line2) {
+            echo "Stream not identical\n";
+            break;
+        }
+    }
+}
+
+function fetchData($stmt, $data, $buffered)
+{
+    if ($buffered) {
+        $result = sqlsrv_fetch($stmt, SQLSRV_SCROLL_FIRST);
+    } else {
+        $result = sqlsrv_fetch($stmt);
+    }
+    if ($result === false) {
+        fatalError('Error when fetching data');
+    }
+
+    // Get image data as a binary stream
+    $image = sqlsrv_get_field($stmt, 0, SQLSRV_PHPTYPE_STREAM(SQLSRV_ENC_BINARY));
+    if ($image === false) {
+        fatalError('Error in calling sqlsrv_get_field');
+    }
+
+    // Does the retrieved stream match with the original?
+    compareStreams($data, $image);
+
+    fclose($image);
+}
+
+function runQuery($conn, $data, $tsql, $buffered, $prepared)
+{
+    if ($prepared) {
+        if ($buffered) {
+            // Pick a random size for the buffer large enough for this test
+            $stmt = sqlsrv_prepare($conn, $tsql, array(), array("Scrollable" => SQLSRV_CURSOR_CLIENT_BUFFERED, "ClientBufferMaxKBSize" => 51200));
+        } else {
+            $stmt = sqlsrv_prepare($conn, $tsql);
+        }
+        if ($stmt === false) {
+            fatalError("Error in preparing the query ($buffered).");
+        }
+        
+        $result = sqlsrv_execute($stmt);
+        if ($result === false) {
+            fatalError("Error in executing the query ($buffered).");
+        }
+    } else {
+        if ($buffered) {
+            // Use the default buffer size in this case
+            $stmt = sqlsrv_query($conn, $tsql, array(), array("Scrollable"=>SQLSRV_CURSOR_CLIENT_BUFFERED));
+        } else {
+            $stmt = sqlsrv_query($conn, $tsql);
+        }
+    
+        if ($stmt === false) {
+            fatalError("Error in sqlsrv_query ($buffered).");
+        }
+    }
+    
+    fetchData($stmt, $data, $buffered);
+}
+
+function runTest($conn, $columnType, $path)
+{
+    $tableName = 'srvTestTable_570' . rand(0, 10);
+    dropTable($conn, $tableName);
+    
+    // Create the test table with only one column
+    $tsql = "CREATE TABLE $tableName([picture] $columnType NOT NULL)";
+    $stmt = sqlsrv_query($conn, $tsql);
+    if (!$stmt) {
+        fatalError("Failed to create table $tableName\n");
+    }
+
+    // Insert php.gif as stream data
+    $tsql = "INSERT INTO $tableName (picture) VALUES (?)";
+    
+    $data = fopen($path, 'rb');
+    if (!$data) {
+        fatalError('Could not open image for reading.');
+    }
+
+    $params = array($data, SQLSRV_PARAM_IN, SQLSRV_PHPTYPE_STREAM(SQLSRV_ENC_BINARY));
+    $stmt = sqlsrv_query($conn, $tsql, array($params));
+    if ($stmt === false) {
+        fatalError("Failed to insert image into $tableName");
+    }
+    do {
+        $read = sqlsrv_send_stream_data($stmt);
+        if ($read === false) {
+            fatalError("Something went wrong trying to insert");
```

Comment:
Won't fatalError be called when running out of  stream data normally? sqlsrv_send_stream_data returns false when no more data is to be sent, but that's not a fatal error.

Reply:
You're right. I'll fix this.